cmd/telemetry-api/main.go

```diff
@@ -60,13 +59,13 @@ func main() {
 		VINVCRepo:  vinvcRepo,
 	}
 
-	privValidator := auth.PrivilegeContractValidator{
-		VehicleNFTAddress: common.HexToAddress(settings.VehicleNFTAddress),
-	}
+	privilegeValidator := auth.NewPrivilegeContractValidator(settings)
```

Comment:
I'm personally trying to stop doing constructors that take in settings. It's too big. Minor thing.